src/settled-physical/CrossMarginPhysicalMath.sol

```diff
@@ -354,45 +354,46 @@ library CrossMarginPhysicalMath {
         //  default value of 0 is not a valid productId so the first run will always set the cache
         uint32 lastUsedProductId;
 
-        for (uint256 i; i < positions.length;) {
-            (, uint32 productId, uint64 expiry,,) = positions[i].tokenId.parseTokenId();
+        unchecked {
+            for (uint256 i; i < positions.length; ++i) {
+                (, uint32 productId, uint64 expiry,, uint64 exerciseWindow) = positions[i].tokenId.parseTokenId();
 
-            // cache product detail if a productId differs from a previous iteration
-            if (productId != lastUsedProductId) {
-                product = _getProductDetails(pomace, productId);
+                // skip expired positions
+                if (expiry < block.timestamp + exerciseWindow) continue;
```

Comment:
I think this doesnt work for physical settled shorts, because if they don't "settle" their account, there is no way to know how much they potentially need deducted from their account.  If the position is a long token, then this is correct because that long is now worthless since they didnt choose to exercise (which is different from cash where longs are technically auto-exercised)